File: app/src/main/java/com/imorning/mediaplayer/activity/BaseActivity.java
package com.imorning.mediaplayer.activity;

import android.Manifest;
import android.os.Bundle;
import android.util.Log;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;

import com.permissionx.guolindev.PermissionX;
import com.permissionx.guolindev.callback.RequestCallback;

import java.util.Arrays;
import java.util.List;

public class BaseActivity extends AppCompatActivity {

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        PermissionX.init(this)
                .permissions(Manifest.permission.READ_EXTERNAL_STORAGE)
                .request(new RequestCallback() {
                    @Override
                    public void onResult(boolean allGranted, @NonNull List<String> grantedList, @NonNull List<String> deniedList) {
                        if (!allGranted) {
                            Log.w(getClass().getSimpleName(), "miss some permission(s): " + Arrays.toString(deniedList.toArray()));
                        }
                    }
                });
    }
}

File: app/src/main/java/com/imorning/mediaplayer/activity/BaseActivity.kt
package com.imorning.mediaplayer.activity

import android.Manifest
import android.os.Bundle
import android.util.Log
import androidx.appcompat.app.AppCompatActivity
import com.imorning.mediaplayer.R
import com.imorning.mediaplayer.utils.ui.ToastUtils
import com.permissionx.guolindev.PermissionX

open class BaseActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        PermissionX.init(this)
            .permissions(Manifest.permission.READ_EXTERNAL_STORAGE)
            .request { _: Boolean, _: List<String?>?, deniedList: List<String?>? ->
                if (deniedList != null) {
                    if (deniedList.contains(Manifest.permission.READ_EXTERNAL_STORAGE)) {
                        if ((this.javaClass.canonicalName)!! == MainActivity::class.qualifiedName) {
                            ToastUtils.showMsg(
                                applicationContext,
                                getString(R.string.permission_warning)
                            )
                        }
                    }
                }
            }
    }

    companion object {
        private const val TAG = "BaseActivity"
    }
}